src/py42/sdk/__init__.py

```diff
@@ -359,7 +365,10 @@ def _init_services(main_connection, main_auth):
         administration=administration_svc,
         archive=ArchiveService(main_connection),
         devices=DeviceService(main_connection),
-        legalhold=LegalHoldService(main_connection),
+        # Only use updated legal hold client if initialized with API Client authorization
```

Comment:
Setting a flag which specifies how the sdk was initialized with the simplest solution I came up with for switching between which client is used - but I'm open to other solutions if there was something else in mind.

Reply:
The general approach is fine but I think I would just use a boolean instead of defining a class/constant for this (I don't anticipate that we will be adding additional authentication methods that will require new API implementations any time soon).